code/test/util.js

```diff
@@ -155,64 +280,467 @@ const createWallet = async ({
+    if (tokenTypes[i] === ONEConstants.TokenType.ERC20) {
+      // await testerc20.transfer(alice.wallet.address, 1000, { from: accounts[0] })
+      await tokenContracts[i].transfer(receiver, tokenAmounts[i], { from: funder })
+      // balances[i]= await tokenAddresses[i].balanceOf(receiver)
+      console.log('FundTokens ERC20')
+    }
+  }
+  if (validate) validatetokenFunding({ receiver, tokenTypes, tokenContracts, tokenAmounts, tokenIds })
+  // return {balances, tids}
+}
 
-  console.log('EVM increased time by', seconds)
+// ==== ADDRESS VALIDATION HELPER FUNCTIONS ====
+// These functions retrieve values using an address
+// They are typically used to validate wallets balances have been funded or updated
+// They do not update State
+
+const validateBalance = async ({ address, amount = HALF_ETH }) => {
+  let balance = await web3.eth.getBalance(address)
+  assert.equal(amount, balance, 'Wallet should have a different balance')
 }
 
-const snapshot = async () => {
-  console.log('Taking EVM Snapshot')
-  return new Promise((resolve, reject) => {
-    web3.currentProvider.send({ jsonrpc: '2.0', method: 'evm_snapshot' },
-      (err, { result } = {}) => {
-        if (err) {
-          reject(err)
-        }
-        resolve(result)
-      })
-  })
+const validatetokenFunding = async ({ receiver, tokenTypes, tokenContracts, tokenAmounts, tokenIds }) => {
+  for (let i = 0; i < tokenTypes.length; i++) {
+    if (tokenTypes[i] === ONEConstants.TokenType.ERC20) {
+      let balanceERC20
+      balanceERC20 = await tokenContracts[i].balanceOf(receiver)
+      assert.equal(tokenAmounts[i], balanceERC20, 'Should have transferred ERC20 tokens to wallet')
+    }
+  }
 }
 
-const revert = async (id) => {
-  console.log(`EVM reverting to ${id}`)
-  return new Promise((resolve, reject) => {
-    web3.currentProvider.send({ jsonrpc: '2.0', method: 'evm_revert', params: [id] }, (err, { result } = {}) => {
-      if (err) {
-        reject(err)
-      }
-      resolve(result)
+// const validateEvents = async ({ tx, opEvent }) => {
+
+// }
+
+// ==== WALLET VALIDATOR HELPER FUNCTIONS ====
+// These functions retrieve values from the current state compare them to oldState
+// They return an updated OldState with the corrected values.
+
+// validateERC20BalanceUpdate
+// const validateERC20BalanceUpdate = async (wallet, oldState, amount) => {
+//   let balanceERC20
+//   let walletBalanceERC20
+//   balanceERC20 = await erc20.balanceOf(wallet.address)
+//   assert.equal(amount, balanceERC20, 'Should have transferred ERC20 tokens to wallet')
+//   walletBalanceERC20 = await wallet.getBalance(ONEConstants.TokenType.ERC20, erc20.address, 0)
+//   assert.equal(amount, walletBalanceERC20, 'Should have tranferred ERC20 tokens to wallet checked via wallet.getBalance')
+// }
+
+// validateUpdateTransaction: changed - nonce, lastOperationTime, commits,
+const validateUpdateTransaction = async ({ wallet, oldState, validateNonce = true }) => {
```

Comment:
This function should not mutate its parameters. Change the name or break it down to two functions?